client/post-editor/editor-diff-viewer/index.jsx

```diff
@@ -88,7 +96,13 @@ class EditorDiffViewer extends PureComponent {
 	};
 
 	recomputeChanges = callback => {
-		const diffNodes = this.node.querySelectorAll( '.text-diff__additions, .text-diff__deletions' );
+		let selectors = '.text-diff__additions, .text-diff__deletions';
+		if ( this.windowWidth > 1040 && this.props.diffView === 'split' ) {
+			selectors =
+				'.editor-diff-viewer__secondary-pane .text-diff__additions,' +
+				' .editor-diff-viewer__main-pane .text-diff__deletions';
```

Comment:
This feels a bit brittle to me, but it looks like we're selecting DOM nodes previously. Maybe be on the lookout to see if there are any alternatives.

Was there a performance penalty to keeping references to the children and then calling https://reactjs.org/docs/react-dom.html#finddomnode ?

Reply:
I think the issue with using references isn't so much performance as the extra complexity they would add to the `EditorDiffViewer`.

We use the list of nodes to measure the position of changes. It lets us scroll the diff view to the first change, and count the number of changes above and below the viewport.

To convert this to refs, we'd probably need to create a ref for each change in the diff array and add it as a property on the change, so we could pass the refs to the `TextDiff` with the rest of the data in the `operations` prop. ([Example](https://github.com/Automattic/wp-calypso/compare/try/post-revisions/split-view-refs#diff-13752d72d22bdbf74b4553a8e8ac22d2).) Then the `TextDiff` could point the refs to its spans. 

We'd only want to do this for the `add` and `del` changes, so we'd need a condition in `EditorDiffViewer` to check for those operations, and a condition in the `TextDiff` component to only set a ref attribute when the change has a `ref` property.

Things start to get much more complicated in the split view, because we'd want to filter out the `add` elements from the left-hand pane and the `del` elements from the right. We may need to do something like divide the node list into two sets.

Trying this out with just the unified view, it looks like that change would replace 7 lines of code with more than 40. The solution for the split view will be even more verbose. I'd say there's something to be said for the simplicity of the DOM API method.